File: internal/model/response.go
package model

type ResponseMessage struct {
	Message string      `json:"message"`
	Meta    MetaMessage `json:"meta"`
}

type ResponseBody struct {
	Response any  `json:"result"`
	Meta     Meta `json:"meta"`
}

type PaginationResponse struct {
	Result any            `json:"result"`
	Meta   PaginationMeta `json:"meta"`
}

type PaginationMeta struct {
	TotalData int    `json:"total_data"`
	Page      int    `json:"page"`
	Size      int    `json:"size"`
	Previous  string `json:"previous"`
	Next      string `json:"next"`
}

type Meta struct {
	Status    string   `json:"status"`
	Method    string   `json:"method"`
	Parameter []string `json:"parameters"`
}

type MetaMessage struct {
	Status string `json:"status"`
	Method string `json:"method"`
}

File: resources/types/response.ts
export interface Response<T> {
    result: T
    meta: {
        status: string
        code: number
    }
}

export interface ResponsePagination<T> {
    result: T
    meta: {
        total_data: number
        total_page: number
        page: number
        size: number
        previous: string
        next: string
    }
}

export interface Pagination {
    page: number
    total_page: number
    previous: string
    next: string
}

export interface MetaData {
    total_data: number
    total_page: number
    page: number
    size: number
    previous: string
    next: string
}